staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -1101,7 +1104,7 @@ func (alloc *allocator) checkAvailableCapacity(device deviceWithID) (bool, error
 			Pool:   slice.Spec.Pool.Name,
 			Device: device.Name,
 		}
-		if !alloc.allocatedDevices.Has(deviceID) && !alloc.allocatingDevices[deviceID] {
+		if !alloc.allocatedDevices.Has(deviceID) && !alloc.allocatingDeviceForAnyClaim(deviceID) {
```

Comment:
First check looks redundant to me as when second (`!alloc.allocatingDeviceForAnyClaim(deviceID)`) is true, first is true as well.